independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.registry.util;
+
+import io.quarkus.maven.ArtifactCoords;
+import io.quarkus.registry.Constants;
+
+public class PlatformArtifacts {
+
+    public static ArtifactCoords bomToCatalog(ArtifactCoords bom) {
+        return bomToCatalog(bom.getGroupId(), bom.getArtifactId(), bom.getVersion());
+    }
+
+    public static ArtifactCoords bomToCatalog(String bomGroupId, String bomArtifactId, String bomVersion) {
+        return new ArtifactCoords(bomGroupId, bomArtifactId + Constants.PLATFORM_DESCRIPTOR_ARTIFACT_ID_SUFFIX, bomVersion,
+                "json", bomVersion);
+    }
+
+    public static ArtifactCoords toCatalog(ArtifactCoords coords) {
```

Comment:
I don't really see what this one bring over the `bomToCatalog`?

Reply:
bomToCatalog doesn't expects the BOM coords as the argument. This method is used when you don't know whether the artifact you have is a catalog one or the bom. There are such cases in the codebase.

Reply:
ah ok sure so `ensureCatalogArtifact` maybe?